public/app/features/datasources/components/picker/DataSourceDropdown.tsx

```diff
@@ -154,14 +164,18 @@ export function DataSourceDropdown(props: DataSourceDropdownProps) {
             <PickerContent
               keyboardEvents={keyboardEvents}
               filterTerm={filterTerm}
-              onChange={(ds: DataSourceInstanceSettings<DataSourceJsonData>) => {
+              onChange={(
+                ds: DataSourceInstanceSettings<DataSourceJsonData>,
+                defaultQueries?: DataQuery[] | GrafanaQuery[]
+              ) => {
                 onClose();
-                onChange(ds);
+                onChange(ds, defaultQueries);
               }}
               onClose={onClose}
               current={currentDataSourceInstanceSettings}
               style={popper.styles.popper}
               ref={setSelectorElement}
+              onClickAddCSV={onClickAddCSV}
               {...restProps}
```

Comment:
All these props are not documented. These props include the `DataSourceList` props that are needed to filter out the element in the picker, and they are very important for the DS use cases: `mixed`, `variables`, etc.

Since they are not documented, when creating the tests we didn't realize they exist. So they are invisible to anyone using the picker across the product.

This is why I don't like to use the spread operator for props 😢

Reply:
In my opinion we should make another PR to move it into a filter object and pass that instead of 10 separate booleans and a filter function. We're not really doing anything with them except passing them to `useDatasources` that passes it to `dataSourceSrv.getList`

We can keep it as separate attributes on the DataSourceDropdown to keep it consistent with the old picker.

Reply:
I agree, but at least they should be documented in the component prop type to indicate they exist.

Then we can refactor them to put them under a single object of `filters` I like that.